tests/integration/test_cloudformation.py

```diff
@@ -1961,40 +1961,46 @@ def test_deploy_stack_with_sub_select_and_sub_getaz(self):
         metric_alarms = cw_client.describe_alarms().get("MetricAlarms", [])
         composite_alarms = cw_client.describe_alarms().get("CompositeAlarms", [])
 
-        # deploy stack
-        create_and_await_stack(StackName=stack_name, TemplateBody=template)
-        exports = cfn_client.list_exports()["Exports"]
+        def _assert_resources_created():
+            exports = cfn_client.list_exports()["Exports"]
+            subnets = [export for export in exports if export["Name"] == "public-sn-a"]
+            instances = [export for export in exports if export["Name"] == "RegmonEc2InstanceId"]
 
-        subnets = [export for export in exports if export["Name"] == "public-sn-a"]
-        instances = [export for export in exports if export["Name"] == "RegmonEc2InstanceId"]
+            self.assertEqual(1, len(subnets))
+            self.assertEqual(1, len(instances))
 
-        self.assertEqual(1, len(subnets))
-        self.assertEqual(1, len(instances))
+            subnet_id = subnets[0]["Value"]
+            instance_id = instances[0]["Value"]
 
-        subnet_id = subnets[0]["Value"]
-        instance_id = instances[0]["Value"]
+            ec2_client = aws_stack.connect_to_service("ec2")
+            resp = ec2_client.describe_subnets(SubnetIds=[subnet_id])
+            self.assertEqual(1, len(resp["Subnets"]))
 
-        ec2_client = aws_stack.connect_to_service("ec2")
-        resp = ec2_client.describe_subnets(SubnetIds=[subnet_id])
-        self.assertEqual(1, len(resp["Subnets"]))
+            resp = ec2_client.describe_instances(InstanceIds=[instance_id])
+            self.assertEqual(1, len(resp["Reservations"][0]["Instances"]))
 
-        resp = ec2_client.describe_instances(InstanceIds=[instance_id])
-        self.assertEqual(1, len(resp["Reservations"][0]["Instances"]))
+            # assert creation of further resources
+            resp = sns_client.list_topics()
+            topic_arns = [tp["TopicArn"] for tp in resp["Topics"]]
+            self.assertIn(aws_stack.sns_topic_arn("companyname-slack-topic"), topic_arns)
+            metric_alarms_after = cw_client.describe_alarms().get("MetricAlarms", [])
+            composite_alarms_after = cw_client.describe_alarms().get("CompositeAlarms", [])
+            self.assertEqual(len(metric_alarms) + 1, len(metric_alarms_after))
```

Comment:
Same nit as above. This assumes sequential test execution + proper cleanup (relies on a predictable global state that is only modified by this test alone). 

Leaving these in for now is potentially fine, but a TODO would be nice so we remember to rework this next time :+1: